docs/stdlib/constraints.rst

```diff
@@ -238,9 +237,9 @@ Constraints
     .. note::
 
         This constraint also has an additional effect of creating an
-        implicit :ref:`index <ref_datamodel_indexes>` on the link or
-        property. This means that in the above example there's no need to
-        add explicit indexes for the ``name`` property.
+        implicit :ref:`index <ref_datamodel_indexes>` on a link or
+        link property. This means that in the ``User`` example there's no need
```

Comment:
I think this was correct as originally written. Not sure if it's possible to constrain on link properties to be honest with you. If it is and if this also holds true for them, I believe that would be in *addition* to properties, rather than *instead* of them.
```suggestion
        property. This means that in the ``User`` example there's no need
```

Maybe @vpetrovykh can confirm?

Reply:
Ah, I just noticed that one is a typo on my part: should say "on a property or link property", not "link or link property".

Reply:
Are you sure that's accurate? This should be about properties and link properties rather than properties and links?

Reply:
A more correct statement would be "an implicit index is created using the expression specified in the `exclusive` constraint", so if we are constraining on a bare property or a link property, that would have an effect of creating an index on it, so you don't have to do this:

```
type Foo {
   name: str { constraint exclusive };
   index on (.name)
}
```

All links are already indexed, so this point is moot for those.

Reply:
Yeah, maybe could use that example as a show-not-tell with inline comments:

```
type Foo {
   name: str { constraint exclusive }; # Makes implicit index
   index on (.name)                          # So you don't need to do this
}
```

Reply:
> Are you sure that's accurate? This should be about properties and link properties rather than properties and links?

Reading this again I think I'll just delete the word link and leave `property` since a link property is also a property and don't want to distract the reader from the subject which is just a quick note on the `exclusive` constraint.

Also I think maybe this will work better: also un-notify the note section and move it up to directly under `type User`. That lets the following section breathe on its own and finally end on the part that you brought up the new issue for yesterday.

https://github.com/edgedb/edgedb/commit/7c017bf37f684cff121124f3cbd6b545447cbaeb

Reply:
Actually...this could work. Move the note above the example and add a commented out line to explicitly show what you don't need to do.

https://github.com/edgedb/edgedb/commit/65763f74aad1b499971a1815e57f360bf7c92c98